src/celeritas/io/RootStepWriter.hh

```diff
@@ -0,0 +1,120 @@
+//----------------------------------*-C++-*----------------------------------//
+// Copyright 2022 UT-Battelle, LLC, and other Celeritas developers.
+// See the top-level COPYRIGHT file for details.
+// SPDX-License-Identifier: (Apache-2.0 OR MIT)
+//---------------------------------------------------------------------------//
+//! \file celeritas/io/RootStepWriter.hh
+//---------------------------------------------------------------------------//
+#pragma once
+
+#include <array>
+
+#include "celeritas_config.h"
+#include "corecel/Assert.hh"
+#include "celeritas/io/RootFileManager.hh"
+#include "celeritas/phys/ParticleParams.hh"
+#include "celeritas/user/StepInterface.hh"
+
+namespace celeritas
+{
+//---------------------------------------------------------------------------//
+/*!
+ * Write "MC truth" data to ROOT at every step.
+ *
+ * `TTree::Fill()` is called for each step and thread id, making each ROOT
+ * entry a step. Since the ROOT data is stored in branches with primitive types
+ * instead of a full struct, no dictionaries are needed for reading the output
+ * file.
+ */
+class RootStepWriter final : public StepInterface
+{
+  public:
+    //!@{
+    //! \name Type aliases
+    using SPRootFileManager = std::shared_ptr<RootFileManager>;
+    using SPParticleParams  = std::shared_ptr<const ParticleParams>;
+    //!@}
+
+    // Construct with RootFileManager and ParticleParams
+    explicit RootStepWriter(SPRootFileManager root_manager,
```

Comment:
Since there's more than one argument you shouldn't make the constructor explicit.